packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -664,6 +664,56 @@ export class KubernetesClient {
     }
   }
 
+  async restartPod(name: string, timeoutMs: number = 60000): Promise<void> {
+    let telemetryOptions = {};
+    try {
+      const ns = this.currentNamespace;
+      const connected = await this.checkConnection();
+      if (ns && connected) {
+        const pod = await this.readNamespacedPod(name, ns);
+        if (pod?.metadata) {
+          const coreApi = this.kubeConfig.makeApiClient(CoreV1Api);
+          // To restart pod we need to delete existed pod and create the new one
+          await coreApi.deleteNamespacedPod(name, ns);
+
+          pod.metadata.resourceVersion = undefined;
+          pod.metadata.creationTimestamp = new Date();
+
```

Comment:
I would recreate a metadata field, with only the expected values: name, namespace, labels, annotations. The other ones (including creationTimestamp) are not expected to be set by the user, but by the system (https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/object-meta/#ObjectMeta)

You can also delete the status field.

Reply:
@feloy I've updated the PR, now new pod configuration doesn't contain status field and the only necessary metadata information